test/boost/sstable_datafile_test.cc

```diff
@@ -1030,17 +1033,18 @@ SEASTAR_TEST_CASE(test_promoted_index_read) {
         auto ck2 = clustering_key::from_exploded(*s, {int32_type->decompose(0), int32_type->decompose(0)});
         auto ck3 = clustering_key::from_exploded(*s, {int32_type->decompose(0), int32_type->decompose(1)});
 
-        auto rd = sstable_reader(sst, s, env.make_reader_permit());
-        using kind = mutation_fragment::kind;
+        auto rd = sstable_reader_v2(sst, s, env.make_reader_permit());
+        using kind = mutation_fragment_v2::kind;
+        // cannot check range tombstone changes more precisely because
+        // the prepared sstables have differing tombstone timestamps
```

Comment:
I'm not sure I understand. What exactly fails here if you enable the checks below?